catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/model/FilterJsonNode.java

```diff
@@ -48,41 +46,38 @@ public FilterNodeMapImpl(final Map<String, Object> json) {
     }
   }
 
-  @Override
   public boolean isLeaf() {
-    return json.get(CHILDREN) == null;
-  }
-
-  @Override
-  public boolean isTemplated() {
-    return json.get(TEMPLATE_PROPERTIES) != null;
+    return !(hasChildren() || isFunction());
   }
 
-  @Override
   public String getOperator() {
     return type;
   }
 
-  @Override
-  public List<FilterNode> getChildren() {
+  public boolean hasChildren() {
+    return json.get(CHILDREN) != null;
+  }
+
+  public boolean isFunction() {
+    return json.get(PROPERTY) instanceof LazyValueMap;
```

Comment:
❓ For now, we won't be able to support functions on `property`, `value`, or **both**?

Reply:
functions are always stored in the property json attribute